File: AtCoder/abc207/C.kt
import kotlin.math.max
import kotlin.math.min

private fun readLn() = readLine() ?: ""
private fun readLns() = readLn().split(" ")
private fun readInt() = readLn().toInt()
private fun readLongs() = readLns().map { it.toLong() }

fun main() {
    val n = readInt()
    val arr = mutableListOf<Pair<Long, Long>>()
    for (i in 0 until n) {
        val nums = readLongs().toMutableList()
        nums[1] *= 2L
        nums[2] *= 2L
        when (nums[0]) {
            1L -> arr.add(Pair(nums[1], nums[2]))
            2L -> arr.add(Pair(nums[1], nums[2] - 1))
            3L -> arr.add(Pair(nums[1] + 1, nums[2]))
            4L -> arr.add(Pair(nums[1] + 1, nums[2] - 1))
        }
    }
    var ans = 0
    for (i in 0 until n) {
        for (j in i + 1 until n) {
            if (max(arr[i].first, arr[j].first) <=
                min(arr[i].second, arr[j].second)) {
                ++ans
            }
        }
    }
    print(ans)
}

File: AtCoder/abc207/C.swift
private func readLn() -> String {
	return readLine() ?? ""
}

private func readLns() -> [String] {
	return readLn().split(separator: " ").map{ String($0) }
}

private func readInt() -> Int {
	return Int(readLn()) ?? 0
}

private func readLongs() -> [Int64] {
	return readLns().compactMap{ Int64($0) }
}

let n = readInt()
var arr = [[Int64]]()
for _ in 0 ..< n {
	var nums = readLongs()
	nums[1] *= 2
	nums[2] *= 2
	if nums[0] == 1 {
		arr.append([nums[1], nums[2]])
	} else if nums[0] == 2 {
		arr.append([nums[1], nums[2] - 1])
	} else if nums[0] == 3 {
		arr.append([nums[1] + 1, nums[2]])
	} else {
		arr.append([nums[1] + 1, nums[2] - 1])
	}
}
var ans = 0
for i in 0 ..< n {
	for j in i + 1 ..< n {
		if max(arr[i][0], arr[j][0]) <= 
           min(arr[i][1], arr[j][1]) {
			ans += 1
		}
	}
}
print(ans)